sympy/polys/ring_series.py

```diff
@@ -753,7 +753,7 @@ def rs_diff(p, x):
     for expv in p:
         if expv[n]:
             e = monomial_ldiv(expv, mn)
-            p1[e] = p[expv]*expv[n]
+            p1[e] = R.domain_new(p[expv]*expv[n])
```

Comment:
`expv` is the key variable for the `PolyElement` (which is `p` here). It's supposed to denote the degree of the polynomial variable. 

For example : 
```
p = {(0, 0):1, (0, 1):-1, (0, 2): -1/2}
```
would mean `1 - x - x**2/2` (when translated to `expr`)
Maybe, the datatype of the key contents should be independent of the field.